azure_functions_worker/bindings/meta.py

```diff
@@ -94,5 +104,37 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
             f'unable to encode outgoing TypedData: '
             f'unsupported type "{binding}" for '
             f'Python type "{type(obj).__name__}"')
+    return datum
 
+
+def to_outgoing_proto(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type]) -> protos.TypedData:
+    datum = get_datum(binding, obj, pytype)
     return datumdef.datum_as_proto(datum)
+
+
+def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
+                              pytype: typing.Optional[type],
+                              out_name: str,
+                              shmem_mgr) \
+        -> protos.ParameterBinding:
+    datum = get_datum(binding, obj, pytype)
+    shared_mem_value = None
+    parameter_binding = None
+    # If shared memory is enabled, try to transfer to host over shared memory
+    if shmem_mgr.is_enabled() and shmem_mgr.is_supported(datum):
+        shared_mem_value = datumdef.Datum.to_rpc_shared_memory(datum, shmem_mgr)
+    # Check if data was written into shared memory
+    if shared_mem_value is not None:
+        # If it was, then use the rpc_shared_memory field in response message
+        parameter_binding = protos.ParameterBinding(
+            name=out_name,
+            rpc_shared_memory=shared_mem_value)
+    else:
+        # If not, send it as part of the response message over RPC
+        rpc_val = datumdef.datum_as_proto(datum)
+        assert rpc_val is not None
+        parameter_binding = protos.ParameterBinding(
```

Comment:
To simplify - can we do returns here, rather than have `parameter_binding = None`:
```suggestion
        return protos.ParameterBinding(
```

Reply:
And remove L140?